sherpa/astro/ui/utils.py

```diff
@@ -174,19 +179,25 @@ def _check_pha_tabstops(data: DataPHA,
     return np.zeros(data.size)
 
 
-def _save_errorcol(session, idval, filename, bkg_id,
-                   clobber, asciiflag,
-                   get_err, colname):
+def _save_errorcol(session,
+                   idval: Optional[IdType],
+                   filename,
```

Comment:
Do we support pathlib? If not (or we don't know for sure) we could just type it as string for now.

```suggestion
                   filename: str,
```

Reply:
So, general pathlib support is 

- #1070

and I'd hoped to get to it, but am not sure I will for 4.17.0.

However, this is covered by a combination of 

- https://github.com/sherpa/sherpa/pull/2114#discussion_r1732743640
- https://github.com/sherpa/sherpa/pull/2114#discussion_r1732781845